spec/negotiation.html

```diff
@@ -96,47 +97,54 @@ <h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-lookupmatcher" aoid="LookupMatcher">
-      <h1>LookupMatcher ( _availableLocales_, _requestedLocales_ )</h1>
-
-      <p>
-        The LookupMatcher abstract operation compares _requestedLocales_, which must be a List as returned by CanonicalizeLocaleList, against the locales in _availableLocales_ and determines the best available language to meet the request. The following steps are taken:
-      </p>
-
+    <emu-clause id="sec-lookupmatcher" type="abstract operation">
+      <h1>
+        LookupMatcher (
+          _availableLocales_: a List of Unicode canonicalized locale identifiers,
+          _requestedLocales_: a List of Unicode canonicalized locale identifiers,
+        ): a Record with fields [[locale]] (a Unicode canonicalized locale identifier) and [[extension]] (a Unicode locale extension sequence or ~empty~)
```

Comment:
I don't think any other spec does optional fields in records. Is `~empty~` an adequate solution or is it actually clearer as optional?

Reply:
I'd say it's an adequate solution, because IMO there isn't much of a difference when [ResolveLocale](https://tc39.es/ecma402/#sec-resolvelocale) performs the test `If r.[[extension]] is not empty, then` instead of `If r has an [[extension]] field, then`.